Ship_Game/Ships/Ship_ModuleGrid.cs

```diff
@@ -526,17 +526,19 @@ public ShipModule HitTestSingle(Vector2 worldHitPos, float hitRadius, bool ignor
                     remainingDamage = diagonalDamage; // start checking from diagonal module
                 }
 
-                if (mq.Module.DamageExplosive(damageSource, ref remainingDamage, mq.Distance)
-                    && mq.Type == DamageTransfer.Root)
-                {
-                    return; // Root module absorbed all the explosion
-                }
 
                 if (mq.Type == DamageTransfer.Root)
                 {
-                    // explosion damage will now be whats left after root module exploded
-                    damageAmount = remainingDamage; 
+                    if (mq.Module.DamageExplosive(damageSource, ref rootDamage))
+                        return; // Root module absorbed all the explosion
+                    else
+                        damageAmount = remainingDamage = diagonalDamage = rootDamage; // This is the new damage after root module exploded

```

Comment:
This is a bit confusing, doesn't it apply the original root damage to the damage amount?

Reply:
actually it should be spread per quadrant here as well, if the root module died and left damage for the explosion. I'll fix this.